### DISCUSSION TITLE
Who is using Nx in production?

### DISCUSSION THREAD

**Discussion by vsavkin** [nrwl/nx#5483] [Show and Tell]
Nx has been growing fast. A lot of companies use Nx today, some of them are listed here: https://nx.dev

If you are using Nx and would like to share this fact with the world, please post your company name with a link in the comment below. We appreciate it.

**Comment by BuckyMaler** [comment]
The development group I'm a part of at [National Oilwell Varco](https://www.nov.com/) has been using Nx for almost a year. We're loving it! Earlier this month we updated to version 8, and are excited about Storybook and the recent caching work. I work on a front-end tooling and infrastructure team with @ZachJW34. We've contributed a few times to the Nx project and look forward to contributing more. Thank you team Nrwl for all your hard work! 👍

**Comment by Sikora00** [comment]
At [ValueAdd](https://valueadd.pl/) (@valueadd-poland) Nx is an important part of our tech stack for over a year and we use it with joy. We deployed a lot of software to production with support from tools from nrwl. We are looking forward to taking advantage of the latest nx features like support for the story book and we are waiting for new, awesome tools in the future.

**Comment by wickstargazer** [comment]
Hi we are using nx in production, our site is [FlowAccount.com](https://flowaccount.com)
Its been some good few months we are dwelling into Nx worskspace. Its a pretty seamless development experience!

And we have wrote our own schematic to work with Nx as well ... its an implementation of serverless node typescript, using nx/angular-cli to generate/serve offline and deploy to aws cloud. 

[@flowaccount/nx-serverless](https://www.npmjs.com/package/@flowaccount/nx-serverless)

The fun part of it is being to work on a full-stack app within one workspace. I have also added tree-shaking for the node_modules and its working well.

We are also adding enhancements and features to it, to help integrate our development workflow and CI/CD pipeline :)

Thankz for a great open-source product!

**Comment by tashoecraft** [comment]
At icims we use Nx to power our Attract product line of career sites: https://www.icims.com/talent-acquisition-software/career-site-solutions/. We power the sites of some of the largest companies, Pepsico, Fedex, Johnson & Johnson, Mcdonalds. Nx serves primarily the searching and applying for jobs portion, applicant data management applications, internal apps, etc. 

It's been very helpful in reusing code quickly without having to deploy to a hosted npm, version tag, and redeploy when changes come down the line. 

Very excited about the future of NX as we pull more applications into the repo.

**Comment by alfaproject** [comment]
We've recently started a company (actually, been almost a year, dayum...) to make a casino platform. I'm pretty sure we are the first ones in the world to have made a 100% serverless casino platform and it would have been very difficult without `nx`, so thank you guys for this amazing tool. I'm just sad I didn't have time to contribute to open source projects this past year, because you know... busy.

We are called The Mill Adventure, https://www.themill.io/

**Comment by LayZeeDK** [comment]
[Systemate A/S](https://www.systemate.dk/?utm_source=github.com/nrwl/nx/discussions/5483&utm_medium=link&utm_campaign=lars) is using Nx in production for several systems in the Danish energy industry, including national critical infrastructure.

A public example is the monorepo for the Danish Green Data Hub and Energy Track and Trace applications with their accompanying design system https://github.com/Energinet-DataHub/greenforce-frontend. The Nx monorepo helps us collaborate across teams, removes the need for versioning our design system, enables us to share tooling including GitHub workflows and actions, and reduces the maintenance cost of our systems.

We're also using Nx for our energy management system Green Imprint (Grønt Aftryk in Danish).

**Comment by Carniatto** [comment]
We are very glad to use it at Riaktr for most of our apps and services. It's has been a ride change for us.
https://www.riaktr.com/

**Comment by dsebastien** [comment]
For Didowi, our upcoming product, we're using Nx as the glue between all the parts of the overall system.

We combine Nx, Angular, NestJS and Kubernetes (also during development) among other things :)

Really glad so far!

**Comment by klaascuvelier** [comment]
We're using it as well at Showpad:
https://www.showpad.com

**Comment by klauvi** [comment]
Sensa
https://sensa.is/

**Comment by yannickglt** [comment]
We use it at Elium https://elium.io

**Comment by stefan-karlsson** [comment]
We use it at MatHem https://mathem.se

**Comment by pf1gura** [comment]
We love it at [Placester](https://placester.com). We started using it about half a year ago as a replacement for `lerna`. The transition was smooth and it enabled us to do so much great stuff without worrying about initial setup. It just works!

**Comment by lithqube** [comment]
I'm really happy that I introduce NX to Zertificon Solutions GmbH - zertificon.com and thank you guys for the great work and support ☺️

**Comment by hotpixelgroup** [comment]
We use it at [Hot Pixel Group](http://www.hotpixelgroup.com). You guys are doing great work!

**Comment by osc707** [comment]
We do at my department @ Adobe. I was hired to architect our AdTech DSP (originally in PHP & angularJS) to Angular (Nx, Ngrx). We have built a few stand alone features over the last 18 months. The last 6 months we are finally doing a platform rebuild with a configuration driven Angular UI leveraging dynamic component generation. [Adobe AdCloud](https://www.tubemogul.com/)

**Comment by demisx** [comment]
## ❤️
@ DimasList.org - https://dimaslist.org

**Comment by ramjawade** [comment]
We are using Nx for our Enterprise Application. It's great tool with well defined architecture for large scale applications with many useful feature like storybook, dependency graph and many more. Looking forward to hear from you guys about Bazel official support.

**Comment by ball6847** [comment]
At [7peaks software](https://7peakssoftware.com/), We develop many enterprise software for our clients in Thailand. We started using NX for almost 2 years now (since nx v0.7) and we still love it. 
We always happy to recommend NX to Angular Community in Thailand.

Thanks for creating NX.

**Comment by tonmanna** [comment]
We have just begun looking into NX at [ITOPPLUS](http://www.itopplus.com). Before we try it on production, we have a lot of questions on how NX can solve our issues. After researching and meeting people that know NX well. After our research, we have no doubts anymore. We're ready to adopt NX on our next production project.
Thanks: @wickstargazer @ball6847 @perjerz. for let me know a great tool.

**Comment by antoinepairet** [comment]
[Medispring](https://app.medispring.be/) is an EHR from Belgium used by more than 1200 doctors everyday

**Comment by oschlegel** [comment]
We are one of several teams working for [MyPorsche](https://connect-portal.porsche.com/) and we are using nx to maintain our frontend code

**Comment by dsln** [comment]
We use it at [SOTI Inc. ](https://www.soti.net/)  in Ontario as part of our shared component library across our suite of products.

Cc: @Cammisuli

**Comment by vmasek** [comment]
We at [FlowUp](https://flowup.cz/en/tech-stack) are using Nx in production alongside with Ionic, Angular and GraphQL to develop multiple "news" related applications and websites (with universal SSR) as part of the news platform for (unnamed) digital network

**Comment by matthewjh** [comment]
We are very happily using it at Tradeweb in our UI monorepo.

https://www.tradeweb.com/

**Comment by charsleysa** [comment]
Kintal Apps Ltd has been using nrwl nx for about 1.5 years for the main product MyMahi https://mymahi.co.nz
EDIT: Now MyMahi Ltd at https://mymahi.com

**Comment by rexebin** [comment]
Encial Group Ltd. We have been using Nx for our trading and management system since begining.

**Comment by jerryorta-dev** [comment]
Spout is using it for a music app that is angular front-end inside of electron and firebase. Nx has been perfect to work with multiple platforms in the same repo. I am utilizing the knowledge from the Advance Workspace course to clean up and organize a lot of code. This has literally saved the app.

**Comment by yharaskrik** [comment]
https://trellis.org is! We have been using Nx for quite a while now! We love it so much a few of us (@Doginal) in the company have contributed to the repo itself and have started a community run Slack workspace for Nx users called [@nrwl/community](https://join.slack.com/t/nrwlcommunity/shared_invite/enQtNzU5MTE4OTQwOTk0LTgxY2E0ZWYzMWE0YzA5ZDA2MWM1NDVhNmI2ZWMyYmZhNWJiODk3MjkxZjY3MzU5ZjRmM2NmNWU1OTgyZmE4Mzc)! We just requested access to the Nx Cloud and are so excited about everything Nx is doing. I couldn't imagine building our company without it, I personally will be using Nx for any project I ever work on.

**Comment by dogmatico** [comment]
We have consolidated some client web applications, 7 at the moment, into a Nx mono-repo at http://bmat.com.

We are happy with the result: having to not manage artifacts and source synchronization has been very satisfying. The only issue, minor because we are a small team, is with commits updating `angular.json` and `nx.json`: I think that modularizing those files can be a really big win and prevent merge conflicts.

**Comment by atifsyedali** [comment]
We (Apployees.com) have been using Nx for ourselves and for another client on a fairly large project. 

Though, we have had to add custom builders, for example, to: (a) a node/web builder to generate a package.json from the dependency graph for each app in the dist folder, (b) a web builder that ports over create-react-app configurations and modifies it for SSR+client, (c) enable distributed caching on redis. We are managing these changes in our own mono-repo that uses Nx to build Nx builders: https://github.com/apployees/apployees-nx

**Comment by myspivey** [comment]
[Shutterstock](https://www.shutterstock.com) is in the process of integrating this to manage our e-commerce code.

**Comment by annsk** [comment]
We use it in our system. 

IC COMPLEX
https://iccomplex.eu

**Comment by LukeHowellDev** [comment]
[Suzy](https://suzy.com/) recently migrated our [Crowdtap](https://crowdtap.com/) application to an NX workspace. It is still just the one application in the workspace as a first step. We recently added our first library and are working on plans to refactor the application into appropriate libraries. Alongside that, we will be pulling in and refactoring our other frontend applications. We are really looking forward to being able to take full advantage of the tooling provided with NX.

**Comment by lukasholzer** [comment]
We @Dynatrace https://www.dynatrace.com/ are heavily depending on you folks! 

You are doing amazing work and help us to focus on our business use cases!

**Comment by alexanderPastornicky** [comment]
[Solar Turbines](https://www.solarturbines.com/) - Thanks guys!

**Comment by hoangleitvn** [comment]
We are an offshoring/outsourcing company based in Vietnam. We have been using this tool for at least 3 projects. The current project we are working on is a big cloud-based SaaS product with Angular, NestJS, Serverless, AWS services.

InnomizeTech https://www.innomizetech.com/

**Comment by GaramNick** [comment]
We at [Oper Credits](https://www.opercredits.com/) have recently transferred our apps to Nx.

Thank you for your hard work & dedication!

**Comment by evtk** [comment]
We are using it at [Achmea](www.achmea.nl), the largest insurance company in the Netherlands. I'm a front-end lead developer, working for one the labels of Achmea: [CentraalBeheer](www.centraalbeheer.nl). At CentraalBeheer, we use NX in our mono repo for all our client-side (and Sitecore) front-end development.

**Comment by bellindj** [comment]
We use it at [84.51°](https://www.8451.com/) on our largest Angular application. NX helped us scale from 1 to 6 teams with 30 active contributors. We also have an NX repo for our Design System (which has a React lib, Angular lib, StencilJS lib, React app and Angular app all under one NX repo!).

**Comment by josecolella** [comment]
We use it at Dynatrace for our Digital business applications. NX has helped us reduce the complexity of sharing higher level libraries between different applications, and making the developer workflow easier with all teams working from the same repository while leveraging modularity, and a libs first approach

**Comment by lukasholzer** [comment]
> We @Dynatrace https://www.dynatrace.com/ are heavily depending on you folks!
> 
> You are doing amazing work and help us to focus on our business use cases!

@josecolella  :D

**Comment by adrael** [comment]
We @betclicgroup are also relying on you. Thanks!
[https://betclic.com](https://betclic.com)

**Comment by rafa-suagu** [comment]
We at @carnovo https://carnovo.com use it and we are very grateful for your great work.

Thanks!

**Comment by plambhsa** [comment]
We're using it for a couple applications at MotivHealth! It's been great!

**Comment by kylecordes** [comment]
We use Nx to manage a big pile of example apps for our Angular training:

https://angularbootcamp.com/

... and also for various complex projects for our customers at:

https://oasisdigital.com/

**Comment by rinckd** [comment]
We use it at Daupler for a variety of apps. Monorepos for the win! [https://daupler.com](https://daupler.com)

**Comment by JustinHoldstock** [comment]
We're currently migrating away from micro-frontends over at [Vendasta](https://www.vendasta.com/) in favour of a monorepo.  Our first few projects are internal apps and our [new Frontend Framework](https://galaxy.vendasta.com/)

**Comment by Niaro** [comment]
We use it at https://www.bridgerpay.com/. Tons of thanks for this awesomeness!

**Comment by aloulouamine** [comment]
We use it at https://nordnet.com.

**Comment by oscarlagatta** [comment]
GOV.UK

**Comment by tytusplanck** [comment]
I recently wrote an article on my [organization's](8451.com) journey of using Nx:

 https://medium.com/@tytusplanck/scaling-our-largest-angular-platform-with-nx-8aa70ee3619f

**Comment by ricmello** [comment]
We use at TOTVS Hospitality.
https://produtos.totvs.com/produto/totvs-hospitalidade/pdv/

I also wrote an article (in Portuguese) a few months ago about how we're using monorepo for 1 year (now are 2 years):
https://medium.com/@ricardo.mello/um-ano-de-monorepo-o-que-aprendemos-e-porque-você-deveria-usá-lo-d3130280bd7d 

And other about Nx itself: https://medium.com/@ricardo.mello/desenvolva-aplicações-full-stack-com-angular-cli-e-nx-2f8dce62975f

**Comment by un33k** [comment]
@vsavkin I have my own consulting company and I've deployed NX for a few clients.  The current one is for an enterprise application (health/insurance)  with 200 active developer (angular) and manages 1/10th of all Americans, which is 35 million people. 

I've just enabled nx cloud for this project, and slowly bumping up the budget cap, as I examine the time saved.

Thank you

**Comment by jschaufele** [comment]
Moving to an opinionated framework is a big step - there can be many value drivers too - My biggest concern is the fact it was originally built for angular, I know it  has scaffolding for other frameworks but most of the people on this feed are using it with angular too - I will be using NextJs - I currently use yarn workspaces but can't find how you pull in existent applications ,  would appreciate insight to these concerns

**Comment by evanfuture** [comment]
We're live at last:

Dungeon Mastery https://dungeonmastery.app

**Comment by un33k** [comment]
@jschaufele - This issue is probably not the best place to address your concern.  But if you had a question on stackoverflow, then perhaps others could provide their experiences with NX, so you can draw your own conclusion.

**Comment by pegaltier** [comment]
I converted my two previous companies:
- https://www.solocal.com/
- https://www.moviplus.ch/

I also used it to generate my Scully boilerplate for my consulting business:
- https://angular-enterprise.com/

**Comment by gullitmiranda** [comment]
https://yube.com.br

**Comment by derbenoo** [comment]
### Using Nx in production with over 1000 libs and apps :rocket: 

We've been using Nx for almost 2 years and we are pretty much all-in at this point. Our entire monorepo evolves around Nx and by now, it would be impossible to keep building our complex SaaS web-application without full-stack TypeScript, the affected graph and library encapsulation. 

Thanks @vsavkin and the team for building such an awesome open-source project :heart: 

If you want to see what is possible with Nx, you can check out:
[https://tapeapp.com](https://tapeapp.com)


Greetings from the Tape team!

**Reply by nelsond6** [reply]
@derbenoo That sounds really impressive, Any chance you could ever do a write up on your ci pipeline setup. Especially around how you determine when to deploy your apps within the mono repo.  We are just starting out our nx monorepo and we can't imagine that many apps and libs yet.  We are still trying to figure out how to setup our ci pipeline to support just 3 apps and a few libs.  We have the building part down, but we are still in the process of determining deployment and testing for each app.

**Comment by Koslun** [comment]
We've been using it in production for over a year with our Angular app along with a Cypress project we've got working but not put into active use yet.

Considerations to start using it for our Node.js server and private npm packages made in TypeScript.

Company: [Normative](normative.io)
Our app: [app.normative.io](app.normative.io)

**Comment by puku0x** [comment]
We use Nx at [Findy](https://findy-code.io/). Thank you very much for your great work!

**Reply by puku0x** [reply]
![](https://img.shields.io/badge/Powered%20by%20Nx-%23143055?logo=nx)
https://shields.io/
https://github.com/simple-icons/simple-icons/pull/4860

**Comment by tashoecraft** [comment]
We use Nx at https://fulcrumpro.com/

**Comment by ashley-hunter** [comment]
We use Nx at [Micro Focus](https://microfocus.com) to build our UI component libraries for Angular, React and Vue all from the same repository while sharing the CSS between all framework implementations. We have also used it for a growing number of enterprise applications too.

Its been working great! Thanks for all the hard work 😊!

**Comment by olsonjj** [comment]
We've been using Nx at [IPRO](https://ipro.com) for almost two years now for our Angular front end.

**Comment by michaelbromley** [comment]
I used Nx to power [this interactive generative art exhibit](https://www.youtube.com/watch?v=QunOupdJyI8) which is now a permanent exhibition at the [National Science Museum](https://www.museoscienza.org/) in Milan, Italy. It was also featured in the [Ars Electronica](https://ars.electronica.art/newdigitaldeal/en/digital-aesthetics/) digital art festival this year.

The app itself is a NestJS server with 3 Angular apps connected to it - a mobile app, admin app and the visual projector. This was my first project using Nx and it was a joy to use!

**Comment by IKatsuba** [comment]
We use Nx at [Tinkoff.ru](https://www.tinkoff.ru). We have about 30 monorepos or more. Also, we have a lot of open-source projects built with Nx:
1. [Taiga UI](https://github.com/TinkoffCreditSystems/taiga-ui)
2. [ng-morph](https://github.com/TinkoffCreditSystems/ng-morph)
3. [microzord](https://github.com/TinkoffCreditSystems/microzord)
4. [micro-sentry](https://github.com/TinkoffCreditSystems/micro-sentry)
5. [shared-library-webpack-plugin](https://github.com/TinkoffCreditSystems/shared-library-webpack-plugin)

And this is just the beginning!

**Comment by caenguidanos** [comment]
Here: [academiares](https://academiares.com)

Thanks!

**Comment by KirillMetrik** [comment]
We have migrated our AngularJS frontend to Angular in NX environment. Our system tracks and manages thousands of cars for our customers.
www.radium.cz

**Comment by nevraydin** [comment]
We've been using it for 2 years at a state bank IT, we have lots of development teams. With Nx, we re-architectured our web framework (using Angular) as a monorepo.  Developers are working fit together and collaborate more closely now. 20+ apps and over 700+ library things (components/pages etc.) in production, it is expected to increase as well.

We think that it would be very hard to scale and maintain this enterprise application without Nx. Thank you for great environment and open-source tool. 😊

**Comment by Niaro** [comment]
@nevraydin wow! How do you manage a complete rebuild? How much time does it take?

**Reply by nevraydin** [reply]
It depends on number of affected projects, when it's too much, it will take almost 10-12 minutes, if not we build in 3 minutes. These durations include lint process (ESlint). If the number of servers are increased, it would be better. However it's acceptable for us.

**Reply by Niaro** [reply]
@nevraydin do u use nx-cloud?)

**Reply by nevraydin** [reply]
No, not yet.

**Comment by javmeister** [comment]
We've been using Nx at Adobe Education Exchange for about 2 years now, because of deployment constraints we have two separate monorepos, one for the backend with 20+ node microservices, thanks to Nx each one as a separate app, and all sharing common code through a lib.

The second monorepo hosts the UI spas, again thanks to NX we can progressively migrate from the older Vue app to the new Web Components (Lit) one step by step, we even managed to share/reuse some web components "back" with the Vue app without complex workarounds or the need to publish them to npm.

Needless to say Nx opened up true scalability for us, and made maintenance of the code a breeze. 

Thanks @vsavkin and the team for all your hard work ❤️

Check out the Adobe Education Exchange: [https://edex.adobe.com](https://edex.adobe.com)

**Comment by dongyuwei** [comment]
I  have migrated from lerna to Nx with one  cmd `npx add-nx-to-monorepo`, and to my surprise, it just works! 
We have 15 react Apps(generated by Create React App ) + 2 react libraries(build by rollup).  Our jenkins CI build is much faster than before if only one App is changed.

One thing I just found is that `nx build:prod myapp` can not run correctly, but `nx build_prod myapp` works as expected, `build:prod` is normal npm scripts in package.json. I use "@nrwl/cli": "12.10.0", not sure if  the issue have been fixed in latest Nx.

And at  this time, we still have lerna as a  dependency, because we still depends on `lerna bootstrap --hoist` to make the npm modules in shared react libraries are hoisted to project root's  node_modules, so Nx can build them without problem. I'm thinking about use one package. json with all dependencies in project root.

**Reply by dongyuwei** [reply]
And `nx dep-graph` helped me  found a circular dependency issue between our two react libraries(which should be one direction depending), which is not found by code review(and our old build tools build them without any warnings ).

**Reply by dongyuwei** [reply]
Update: I've migrated from npm to pnpm, and removed lerna (and the lerna bootstrap --hoist behavior) completely. 
pnpm + Nx works great!

I keep using one package.json per project and removed all package-lock.json(replaced by single pnpm-lock.yaml under the root of our project). pnpm is much better than npm.

**Comment by rubiagatra** [comment]
We are using Nx in our new workflow.

Even we create comparison on hybrid mobile development on it https://github.com/zero-one-group/ionic-react-native-flutter

Also, we created an open-source project using Nx @vsavkin https://github.com/zero-one-group/fxl.js

**Comment by jerryorta-dev** [comment]
A small side project app. Nx provides a great scalable workflow even for small apps that may evolve into something more.

https://quizkit.us/

**Comment by dtslvr** [comment]
We are using Nx in production for [Ghostfol.io](https://ghostfol.io), an open source web app to manage your personal finance based on Nest.js and Angular.

**Comment by antoinepairet** [comment]
[Rosa](https://pro.rosa.be) is a health-tech non-profit run as a startup which is based in Belgium. We use Nx for our entire codebase: frontends, backends, serverless. We will add a mobile app later this year.
kudos for the amazing value Nx provides 💪🏻 😍

**Comment by dim0627** [comment]
I'm using in https://gofood.co.jp/ with React Native app, and NestJS API stack!
Nx is so awesome tool, I'm lovin' it ❤️

**Comment by javmeister** [comment]


**Reply by pegaltier** [reply]
@javmeister you should check how to build on #holochain instead of solana.

**Reply by pegaltier** [reply]
Holochain is actually the evolution of blockchain.

**Reply by pegaltier** [reply]
No problem, I hope you will enjoy your solana choice : 
<img width="635" alt="image" src="https://user-images.githubusercontent.com/2479323/165508458-8c54bd45-c8b7-427a-83e0-0e83e8caf0d6.png">

source: https://twitter.com/Justin_Bons/status/1514646388688232456

**Reply by javmeister** [reply]
Can you please stop hijacking my post?

I am deleting and reposting, to get rid of this, I apologize to everybody else in advance for the spamming

**Reply by javmeister** [reply]
Nx team (@vsavkin) please feel free to delete this entire comment thread to get rid of the spam

**Reply by pegaltier** [reply]
This is not a spam at all, it was intended to be a real discussion but you turned to say i'm shameless and now you want to hide something, maybe the truth?

**Reply by javmeister** [reply]
Ok back to work, I can't continue wasting time, blocked.

**Comment by javmeister** [comment]
Second project we are using Nx for, this time is a Solana game https://singularityonline.app/. This one is even more fun, and it is composed of several apps:

- A main Angular SPA
- A library of Web Components written in Lit, one of them wrapping a Threejs 3D visualization of an entire Galaxy.
- A Firebase functions project for serverless backend
- A Nest project to wrap up some blockchain interaction logic that can only be done from the backend (for security reasons)
- And a shared code library, for data classes, types, interfaces, even a Unity Engine WebGL loader code.
- A @nxrs/cargo project to add Rust programs (smart contracts) to soon replace the firebase functions serverless backend with a fully decentralize equivalent.

The project itself is a browser based Blockchain space MMOG that we will be launching soon, NX allowed us to have a prototype up and running in barely two weeks of work, it just makes scalability so easy.

The development setup is extremely straightforward, and that is and incredible thing to say when you are talking about a game made in Unity with interoperability with the web browser, and the Phantom wallet browser plugin, through a modern Web app.

The only tough choice we ever had to make was React or Angular, and we are actually going to migrate to Web Components + Mobx instead, to reduce the bloat, but since we started organizing the original prototype code into shared libraries and simple components this migration should not cost us that much, again thanks to Nx.

Keep up the amazing work ❤️🚀

**Comment by ali-kamalizade** [comment]
We are using NX at [Sunhat](https://github.com/sunhat-tech) - kind of a given since we had used it at my former employer as well.

**Comment by jerryorta-dev** [comment]
Spout Software is using NX - https://spoutstudio.io/home

The main frameworks are:
* Angular
* Angular Material
* NgRX
* D3
* Electron
* Firebase
* Backblase B2

I have apps for electron, marketing site, authentication, and firebase is in the same repo.

Couldn't have done it without NX.

**Comment by leandrocrs** [comment]
We are using in here [Lista de Presentes DoMeuJeito](https://www.domeujeito.com.br) by [Lojas Riachuelo](https://www.riachuelo.com.br) with React app and web libraries.

**Reply by guisperandio** [reply]
Mto interessante saber que estao usando nx no brasil tb. Parabens, sabia escolha. Como tem sido a experiencia Leandro?

**Comment by flexbox** [comment]
We are using Nx at [Mahalo banking](https://www.linkedin.com/company/mahalobanking/mycompany/).

It's a nice tool in a team of 8+ devs

**Comment by kitsune7** [comment]
We're using Nx at [Weave](https://www.getweave.com/) for our application code (marketing does their own thing on the page that's linked). It's been a tough process, but we ported over thousands of files into an Nx monorepo.

By carefully organizing our feature sets into individual projects and publishing them as internal npm packages, we've been able to rewrite or port cleaned up segments of code to use in a new version of our product without needing to drop support for the product currently being used by customers.

It's been really good for programmer morale to be able to jump straight to a clean (and fast) architecture for new development and demote the old architecture to maintenance as needed while we slowly replace it.

**Comment by cwoebker** [comment]
We at @rembergtech 🐻  in Munich 🇩🇪  are using Nx monorepo in a team of 20+ engineers since the beginning of this year. We use it to manage our frontend, backend microservices, and all of our shared packages.

Main tech used: Angular, Express, Nest.js, Ionic (for mobile applications), & Cypress

Thanks for all of your hard work.

**Comment by busla** [comment]
Icelands governmental digital sercices are all built and managed in a large NX monorepo, consisting of over 300 React/Next/Node/Nest apps/libs.

The development is distributed to around 10  dev teams (around 40 developers) and 1 platform team responsible for CI/CD and the developer platform.

All our code is open-source and can be found here.

https://island.is/en

https://github.com/island-is/island.is

**Comment by Itrulia** [comment]
Education First/Cultural Care uses Nx (previous employer): https://aupair2.culturalcare.com


We.Publish uses Nx: https://wepublish.ch  and  http://github.com/wepublish/wepublish/

**Comment by qortex** [comment]
We do at Qontrol ! https://www.qontrol.io

Angular + NestJs & GraphQL (+ legacy ruby)

**Comment by phihochzwei** [comment]
We use it at https://www.deichmann.com and https://www.snipes.com for currently 15 countries and more will follow

**Comment by lildesert** [comment]
We use it at [teamout](https://www.teamout.com/).
Remix and node apps using typescript.

**Comment by huv1k** [comment]
We are using Nx in [Productboard](https://www.productboard.com/) more then 720+ libs and 1.1+ millions lines of Typescript and 5 production applications + multiple Cloudflare workers 🙌

**Comment by vicb** [comment]
I'm using Nx for flyxc.app (https://github.com/vicb/flyxc)

The build was ~3mn before I switched and went down to a few seconds making the DX much more pleasant.
Nx is getting better release after release.
Before Nx my build was broken ~2 times a year

**Comment by simply10w** [comment]
We use it at [Element451](https://element451.com) for almost 7 years (Dec 17, 2017 when we created our monorepo repository).

**Comment by SerkanSipahi** [comment]
We are using it for 3 1/2 years now!
company name: www.avodaq.com (300 employees and 30 of them are in the development department)

**Comment by santoshyadavdev** [comment]
We are using it at Celonis for our biggest product. 
 https://www.celonis.com/platform/process-mining/

**Comment by thedevtoni** [comment]
We have been using it for three years now in https://orq.ai/

**Comment by tuckerjt07** [comment]
I've been using it for three or four years at Walmart

**Comment by naharoo** [comment]
Hi. We've been using Nx at https://www.newlevelwork.com for a year now. There are 6 developers in our team.

**Comment by dwelch2344** [comment]
We've got several NX projects at @herodevs, including our [CLI](https://github.com/herodevs/cli) and the system that runs our Never-Ending Support initiative. We also work with dozens of orgs who rely on NX for their entire codebase. Much love to the NX team!

**Comment by govindappaarun** [comment]
We have been using at [Amadeus](https://amadeus.com/en ) about 2 years now

**Comment by shhdharmen** [comment]
I have been using it for [angular-material.dev](https://angular-material.dev/home) for almost 10 months now.
